crates/runtime/src/guest_type.rs

```diff
@@ -101,6 +124,9 @@ impl<'a, T> GuestType<'a> for GuestPtr<'a, T> {
     }
 
     fn read(ptr: &GuestPtr<'a, Self>) -> Result<Self, GuestError> {
+        let _ =
+            ptr.mem()
+                .validate_size_align(ptr.offset(), Self::guest_align(), Self::guest_size())?;
```

Comment:
I think this is already happening in `.read()` below, so it may not need to happen here too?

Reply:
Correct, I missed that.